quadriga/src/main/java/edu/asu/diging/quadriga/core/service/impl/CollectionManagerImpl.java

```diff
@@ -97,17 +106,31 @@ public Collection editCollection(String id, String name, String description, Lis
      * @see edu.asu.diging.quadriga.core.service.CollectionManager#deleteCollection(java.lang.String)
      */
     @Override
-    public void deleteCollection(String id) throws CollectionNotFoundException, InvalidObjectIdException {
+    public void deleteCollection(String id,SimpleUser simpleUser) throws CollectionNotFoundException, InvalidObjectIdException,SecurityException {
         Collection collection = findCollection(id);
-        
         if(Objects.nonNull(collection)) {
-            
+
             // Once networks are linked with collections, only empty collections will be deleted
             // If it is linked to a network, we will archive the collection.
+            if (!collection.getOwner().equals(simpleUser.getUsername())) {
+                //If someone other than the owner tries to delete,an exception is thrown
+                throw new SecurityException();
```

Comment:
I don't think that's the right exception to throw here, that's a core java exception I believe and a runtime exception. Just create your own.